project_task_tags/views/project_task.xml

```diff
@@ -0,0 +1,18 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<odoo>
+    <record id="view_task_form2" model="ir.ui.view">
+        <field name="name">project.task.allowed_tags.form.inherit</field>
+        <field name="model">project.task</field>
+        <field name="inherit_id" ref="project.view_task_form2" />
+        <field name="arch" type="xml">
+            <field name="tag_ids" position="before">
+                <field name="allowed_tag_ids" widget="many2many_tags" invisible="1" />
+            </field>
+            <field name="tag_ids" position="attributes">
+                <attribute name="domain">[('id','in',allowed_tag_ids)]</attribute>
+                <!-- make sure there is no project key in context -->
+                <attribute name="context">{}</attribute>
```

Comment:
Why, could this break some functionality ?

Reply:
it is used in name_search and returns all tags linked to proj: https://github.com/odoo/odoo/blob/16.0/addons/project/models/project.py#L2755

Reply:
hmm so what's the point of this module? :thinking: 
Besides, that method is one of the ugliest I've seen in terms of clarity :sweat_smile:

Reply:
the point is to show only tags which are already chosen on parent project. OR all tags available if there is no project.
if we keep project_id in ctxt, doesnt matter what kind of domain is on the field - it always shows more/all tags as a possibility to choose from. So we need to remove proj_id key from context and set domain on field - to show only tags from linked project or all tags (either by computed field, which was here before, or domain proposed by you). Domain alone wont work.